### DISCUSSION TITLE
BaseDataset:   self.num_classed

### DISCUSSION THREAD

**Discussion by zhanggj821** [open-mmlab/mmaction2#1242] [Q&A] [answered]
how do i get the self.num_classes in the file of mmaction/datasets/base.py. I got a None type if I directly use 'self.num_classses'

**Comment by dreamerlin** [comment, marked as answer]
you should set it in config files.

**Reply by zhanggj821** [reply]
thanks for your reply,  but config files do not need to set it normally, i think it may be passed in function "build_from_cfg" but I don't got it.....
I mean if I want to modified the function 'evaluate'  in class 'BaseDataset' i.e., the file of mmaction/datasets/base.py and I wanna use 
the self.num_classes property, but actually I get a None Type

**Reply by dreamerlin** [reply]
https://github.com/open-mmlab/mmaction2/blob/master/configs/recognition/tsn/tsn_r101_1x1x5_50e_mmit_rgb.py#L91-L111

**Reply by zhanggj821** [reply]
thank you. i figure out it，I mistakenly thought it would calculate automatically
